pkg/deploy/lattice/iamauthpolicy_manager.go

```diff
@@ -0,0 +1,89 @@
+package lattice
+
+import (
+	"context"
+	"fmt"
+
+	pkg_aws "github.com/aws/aws-application-networking-k8s/pkg/aws"
+	model "github.com/aws/aws-application-networking-k8s/pkg/model/lattice"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/vpclattice"
+)
+
+type IAMAuthPolicyManager struct {
+	Cloud pkg_aws.Cloud
+}
+
+func (m *IAMAuthPolicyManager) Put(ctx context.Context, policy model.IAMAuthPolicy) (model.IAMAuthPolicyStatus, error) {
+	res, err := m.findResouce(ctx, policy)
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	req := &vpclattice.PutAuthPolicyInput{
+		Policy:             &policy.Policy,
+		ResourceIdentifier: &res.Id,
+	}
+	resp, err := m.Cloud.Lattice().PutAuthPolicyWithContext(ctx, req)
+	if err != nil {
+		return model.IAMAuthPolicyStatus{}, err
+	}
+	return model.IAMAuthPolicyStatus{
+		Resource: res,
+		State:    *resp.State,
+	}, nil
+}
+
+func (m *IAMAuthPolicyManager) Delete(ctx context.Context, policy model.IAMAuthPolicy) error {
+	res, err := m.findResouce(ctx, policy)
+	if err != nil {
+		return err
+	}
+	req := &vpclattice.DeleteAuthPolicyInput{
+		ResourceIdentifier: &res.Id,
+	}
+	_, err = m.Cloud.Lattice().DeleteAuthPolicyWithContext(ctx, req)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *IAMAuthPolicyManager) findResouce(ctx context.Context, policy model.IAMAuthPolicy) (model.IAMAuthPolicyResource, error) {
+	empty := model.IAMAuthPolicyResource{}
+
+	switch policy.Kind {
+
+	case model.GatewayAuthPolicy:
+		out, err := m.Cloud.Lattice().FindServiceNetwork(ctx, policy.ResourceName, "")
+		if err != nil {
+			return empty, err
+		}
+		arn, err := arn.Parse(*out.SvcNetwork.Arn)
+		if err != nil {
+			return empty, err
+		}
+		return model.IAMAuthPolicyResource{
+			Name: policy.ResourceName,
+			Id:   *out.SvcNetwork.Id,
```

Comment:
If this is specifically the service network id, perhaps Id should be changed to ServiceNetworkId